test/e2e-go/features/catchup/catchpointCatchup_test.go

```diff
@@ -243,7 +253,7 @@ func TestBasicCatchpointCatchup(t *testing.T) {
 	status, err := awaitCatchpointCreation(primaryNodeRestClient, &fixture, 3)
 	a.NoError(err)
 
-	log.Infof("primary node latest catchpoint - %s!\n", status.LastCatchpoint)
+	log.Infof("primary node latest catchpoint - %s!\n", *status.LastCatchpoint)
 	_, err = secondNodeRestClient.Catchup(*status.LastCatchpoint)
 	a.NoError(err)
 
```

Comment:
@algorandskiy I confirm the test regularly fails prior to the PR though I see _periodic_ failures with the PR's changes.  Unfortunately, understanding the root cause is beyond my grasp.  I'm leaving the note in case others have ideas.

Test failure:
```
    fixture.go:119:
                Error Trace:    catchpointCatchup_test.go:266
                Error:          Received unexpected error:
                                timeout waiting for round 21 with last round = 1
                Test:           TestBasicCatchpointCatchup
```
I've attached test output + associated 2nd node logs from a local test pass and a local failure:
* Passing
  * [passing.txt](https://github.com/algorand/go-algorand/files/9305595/passing.txt)
  * [node.log](https://github.com/algorand/go-algorand/files/9305627/node.log)
* Failing
  * [failing.txt](https://github.com/algorand/go-algorand/files/9305596/failing.txt)
  * [node.log](https://github.com/algorand/go-algorand/files/9305623/node.log)

Running `grep catchpoint node.log` produces varying output.  Suggests to me that the node logs provide a clue, but I'm unable to put it together.

Reply:
Thank you Michael, this is
```
    testingLogger.go:38: time="2022-08-10T21:53:58.742544 -0400" level=info msg="primary node latest catchpoint - 16#WXFXAMT2TT6K4FQZ7F54A5XFA2IDHY6WPN5J5JQCLLAACRLDEXDQ!\n" file=catchpointCatchup_test.go function=github.com/algorand/go-algorand/test/e2e-go/features/catchup.TestBasicCatchpointCatchup line=256
```
so instead of round 20 we got catchpoint for 16, and this one requires block 2 in history. I'll add a loop waiting for a specific specific catchpoint (or greater).

Reply:
@algorandskiy Ah got it - thanks for the explanation, it helps to build my intuition.

I'll hold thread open for a change.

Reply:
pushed, please take another look

Reply:
@algorandskiy Thanks for supplying a fix - I agree https://github.com/algorand/go-algorand/commit/c4dacb8fa2c1adae9ac18ae4032b7156ced3dfbb addresses the flaky behavior.

I'm resolving, but for completeness, I leave a context note for your consideration:
* Based on my understanding, https://github.com/algorand/go-algorand/commit/c4dacb8fa2c1adae9ac18ae4032b7156ced3dfbb removes the need to later invoke `awaitCatchpointCreation` because the for-loop accesses the same information acquired by `awaitCatchpointCreation`. 
* So, it's unclear to me if you intentionally left `awaitCatchpointCreation` as is.  I confirmed through local testing that the test continues passing when replacing `awaitCatchpointCreation` with status from the added for-loop.
* Optionally and if you agree, I'd consider removing  `awaitCatchpointCreation` because it's an unnecessary test step.

Reply:
>Based on my understanding, https://github.com/algorand/go-algorand/commit/c4dacb8fa2c1adae9ac18ae4032b7156ced3dfbb removes the need to later invoke awaitCatchpointCreation because the for-loop accesses the same information acquired by awaitCatchpointCreation.

yes, I removed and running a test locally, did not notice it originally.